program_intent_engagement/apps/api/permissions.py

```diff
@@ -0,0 +1,20 @@
+"""
+Permissions for program intent API
+"""
+from rest_framework.permissions import BasePermission
+
+
+class ProgramIntentUserPermissions(BasePermission):
+    """
+    Permission class that checks that user interacts with their data only
+    """
+    def has_permission(self, request, view):
+        if request.method == "POST":
+            modified_user = request.data.get("user")
+            if modified_user:
+                return request.user.lms_user_id == int(modified_user)
+
+    # todo: is this necessary? what does this do?
+    def has_object_permission(self, request, view, obj):
```

Comment:
question: I had added this when referencing the example code from Andy from demographics. That being said, I wasn't sure if this part was necessary?

Reply:
`has_object_permission` allows you to check more fine-grained permissions on objects, like a model instance; in this case, that would be an instance of `ProgramIntent`. `has_permission` defines global permissions that run on all requests, whereas `has_object_permission` defines permissions that run on objects.

For example, you might want `has_permission` to prevent users whose first name is Michael from accessing your API, because they're just bad people. Alternatively, you might want `has_object_permission` to only prevent users under the age of 18 from POSTing `ProgramIntents`, because they're too young to know what they want in life, but they should be able to do everything else.

I _feel_ like what you have is more of an object-level permission, but I think either global-level or object-level permissions will work here. I think you will have to explicitly call `self.check_object_permissions(request, obj)` from your view, because you're not using a generic view, if you decide to use the object-level permissions.